tests/bluetooth/df/connectionless_cte_chains/src/test_remove_cte_from_chain.c

```diff
@@ -152,14 +158,305 @@ void test_remove_cte_from_single_pdu_chain(void)
+	zassert_equal(err, 0,
+		      "Unexpected error while enabling CTE for periodic avertising chain, err: %d",
+		      err);
+
+	/* Swap PDU double buffer and get new latest PDU data */
+	pdu_new = lll_adv_sync_data_latest_get(adv->lll.sync, NULL, &upd);
+	zassert_not_equal(pdu_new, NULL,
+			  "Unexpected value of PDU pointer after PDU double buffer swap");
+
+	pdu_prev = lll_adv_sync_data_peek(adv->lll.sync, NULL);
+	zassert_equal(pdu_prev, pdu_new,
+		      "Unexpected value of PDU pointer after PDU double buffer swap");
+
+	/* Free PDUs fifo should hold single PDU released during double buffer swap. The PDU
+	 * was allocated during advertising set creation.
+	 */
+	pdu_fifo_cnt = lll_adv_free_pdu_fifo_count_get();
+	zassert_equal(pdu_fifo_cnt, TEST_PER_ADV_SINGLE_PDU,
+		      "Unexpected number of free PDUs in a fifo: %d", pdu_fifo_cnt);
+
+	/* Expected free PDUs count is decreased by:
+	 * - single PDU allocated during advertising set creation,
+	 * - number of PDUs allocated for per. adv. chain to Tx CTE
+	 */
+	pdu_mem_cnt_expected -= (TEST_PER_ADV_CHAIN_LENGTH + TEST_PER_ADV_SINGLE_PDU);
+	pdu_mem_cnt = lll_adv_pdu_mem_free_count_get();
+	zassert_equal(pdu_mem_cnt, pdu_mem_cnt_expected,
+		      "Unexpected amount of free PDUs memory: %d, expected %d", pdu_mem_cnt,
+		      pdu_mem_cnt_expected);
+
+	err = ll_df_set_cl_cte_tx_enable(handle, false);
+	zassert_equal(err, 0,
+		      "Unexpected error while disabling CTE for periodic avertising chain, err: %d",
+		      err);
+	/* Validate result */
+	common_validate_per_adv_chain(adv, TEST_PER_ADV_CHAIN_LENGTH);
+
+	/* Swap PDU double buffer to check correctness of release former PDUs */
+	pdu_new = lll_adv_sync_data_latest_get(adv->lll.sync, NULL, &upd);
+	zassert_not_equal(pdu_new, NULL,
+			  "Unexpected value of PDU pointer after PDU double buffer swap");
+
+	/* Validate number of released PDUs */
+
+	/* Number of free PDUs in a fifo is a number of released PDUs from former periodic
+	 * advertising chain. One free PDU that had been in a fifo was used for allocation of
+	 * the new PDU without CTE.
+	 */
+	pdu_fifo_cnt = lll_adv_free_pdu_fifo_count_get();
+	zassert_equal(pdu_fifo_cnt, TEST_PER_ADV_CHAIN_LENGTH,
+		      "Unexpected number of free PDUs in a fifo: %d", pdu_fifo_cnt);
+
+	/* Number of free PDUs in memory pool decreased. Single PDU for AUX_SYNC_IND was acquired
+	 * from free PDUs fifo. TEST_PER_ADV_CHAIN_LENGTH - 1 PDUs were allocated from free PDUs
+	 * memory pool.
+	 */
+	pdu_mem_cnt_expected -= (TEST_PER_ADV_CHAIN_LENGTH - TEST_PER_ADV_SINGLE_PDU);
+	pdu_mem_cnt = lll_adv_pdu_mem_free_count_get();
+	zassert_equal(pdu_mem_cnt, pdu_mem_cnt_expected,
+		      "Unexpected amount of free PDUs memory: %d, expected %d", pdu_mem_cnt,
+		      pdu_mem_cnt_expected);
+	/* Teardown */
+	common_release_per_adv_chain(adv);
+	common_release_adv_set(adv);
+	lll_adv_init();
+}
+
+void remove_cte_from_chain_length_update_after_enqueue_to_lll(uint8_t updated_chain_length)
```

Comment:
I assume that it is by design, but just making sure:
The added functions here share a lot of common code, which could be put into smaller functions for reusability. Was this considered?

Reply:
I have refactored the tests cases.